src/pokemon.c

```diff
@@ -5934,6 +5934,7 @@ bool8 PokemonUseItemEffects(struct Pokemon *mon, u16 item, u8 partyIndex, u8 mov
              && gMain.inBattle && battlerId != MAX_BATTLERS_COUNT && (gBattleMons[battlerId].status2 & STATUS2_CONFUSION))
             {
                 gBattleMons[battlerId].status2 &= ~STATUS2_CONFUSION;
+                gStatuses4[battlerId] &= ~STATUS4_INFINITE_CONFUSION;
```

Comment:
There are other cases of healing infinite confusion that need to be taken care of, like gaining Own Tempo via Skill Swap or Persim Berry for example. I recommend a new function:
```c
void RemoveConfusionStatus(u8 battlerId)
{
    gBattleMons[battlerId].status2 &= ~STATUS2_CONFUSION;
    gStatuses4[battlerId] &= ~STATUS4_INFINITE_CONFUSION;
}

Reply:
persim berry should already be handled by `HOLD_EFFECT_CURE_CONFUSION`. will check for the other cases.